packages/browser-wallet-api/src/wallet-api.ts

```diff
@@ -4,13 +4,35 @@ import {
     EventType,
     MessageType,
 } from '@concordium/browser-wallet-message-hub';
+import { AccountTransactionPayload, AccountTransactionType } from '@concordium/web-sdk';
+import { stringify } from './util';
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 type WalletEventHandler<T = any> = (payload: T) => void;
 
 export interface IWalletApi {
     addChangeAccountListener(handler: WalletEventHandler<string>): void;
-    sendTransaction(): Promise<boolean>;
+    /**
+     * Sends a transaction to the Concordium Wallet and awaits the users action. Note that a header is not sent, and will be constructed by the wallet itself.
+     * Note that if the user rejects signing the transaction, this will throw an error.
+     * @param type the type of transaction that is to be signed and sent.
+     * @param payload the payload of the transaction to be signed and sent.
+     * @param parameters parameters for the initContract and updateContract transactions in JSON-like format.
+     * @param schema schema used for the initContract and updateContract transactions to serialize the parameters. Should be base64 encoded.
+     */
+    sendTransaction(
+        type: AccountTransactionType,
+        payload: AccountTransactionPayload,
+        parameters: Record<string, unknown>,
+        schema: string
```

Comment:
Should this be optional, in case the schema is embedded in the contract?

Reply:
I figured we would require it, because we don't support the embedded schema initially (I guess), but maybe it's better to make it optional now, to avoid changing the API later?

Reply:
Hmm... I guess as long as it goes from required to optional and not the other way around, it doesn't really matter? Wouldn't be a breaking change to make it optional as far as I see.